packages/node_modules/@ciscospark/internal-plugin-wdm/test/unit/spec/device.js

```diff
@@ -3,116 +3,106 @@
-          .then(() => assert.becomes(device.getWebSocketUrl(), `wss://mercury-connection-a5.wbx2.com/v1/apps/wx2/registrations/WEBSOCKETID/messages`));
-      });
-    });
+    describe(`when web-ha-messaging is enabled`, () => {
+      beforeEach(`enable web-ha-messaging`, () => {
+        const deviceConfig = cloneDeep(deviceFixture);
+        deviceConfig.features.developer.push({
+          key: `web-ha-messaging`,
+          val: `true`,
+          value: true,
+          mutable: true,
+          lastModified: `2015-06-29T20:02:48.033Z`
+        });
+        spark.internal.device.set(deviceConfig);
 
-    describe(`#useServiceCatalogUrl`, () => {
-      it(`replaces current url with url from the service catalog`, () => {
-        return assert.becomes(device.useServiceCatalogUrl(`https://conv-a.wbx2.com/conversation/v1`), `https://conv-a4.wbx2.com/conversation/v1`);
-      });
-    });
+        device = spark.internal.device;
 
-    describe(`#clear()`, () => {
-      it(`does not remove \`logger\``, () => {
-        assert.property(device, `logger`);
-        assert.isDefined(device.logger);
-        device.clear();
-        assert.property(device, `logger`);
-        assert.isDefined(device.logger);
+        device.config.preDiscoveryServices = {};
       });
 
-      it(`clears all features`, () => {
-        assert.isAbove(device.features.developer.length, 0);
-        device.clear();
-        assert.lengthOf(device.features.developer, 0);
-      });
-    });
+      describe(`#_updateServiceCatalog`, () => {
+        it(`refreshes the device if serviceHostMap is not defined and ha is enabled`, () => {
+          device.features.developer.set([{
+            key: `web-ha-messaging`,
+            val: `true`,
+            value: true,
+            mutable: true,
+            lastModified: `2015-06-29T20:02:48.033Z`
+          }]);
+          device.services = {
+            validServiceUrl: `http://example.com/valid-service`,
+            anotherServiceUrl: `http://another.com/another-service`
+          };
+          sinon.spy(device, `refresh`);
+          device.serviceHostMap = undefined;
 
-    describe(`#determineService()`, () => {
-      it(`determines the service backing the specified url`, () => assert.becomes(device.determineService(`https://conv-a.wbx2.com/conversation/api/v1/conversations`), `conversation`));
-      it(`rejects if no service can be determined`, () => assert.isRejected(device.determineService(`https://rogue-service.example.com`), /does not reflect a known service/));
-    });
+          assert.calledOnce(device.refresh);
+        });
 
-    describe(`#getPreDiscoveryServiceUrl()`, () => {
-      it(`requires a \`service\` parameter`, () => {
-        return assert.isRejected(device.getPreDiscoveryServiceUrl(), /`service` is a required parameter/);
-      });
+        it(`updates the service catalog`, () => {
+          device.serviceCatalog.reset();
```

Comment:
what does this do?

Reply:
this clears the serviceCatalog. it was to test that serviceCatalog.set was called for the 2 items in hostCatalog `assert.isTrue(device.serviceCatalog.length === 2)`